src/Sylius/Bundle/ApiBundle/CommandHandler/RegisterShopUserHandler.php

```diff
@@ -66,18 +81,30 @@ public function __invoke(RegisterShopUser $command): void
         $customer->setPhoneNumber($command->phoneNumber);
         $customer->setUser($user);
 
-        $this->handleVerification($user);
-
-        $this->shopUserManager->persist($user);
-    }
-
-    private function handleVerification(ShopUserInterface $user): void
-    {
         /** @var ChannelInterface $channel */
-        $channel = $this->channelContext->getChannel();
+        $channel = $this->channelRepository->findOneByCode($command->channelCode);
 
-        if (!$channel->isAccountVerificationRequired()) {
+        $this->shopUserManager->persist($user);
+        $this->shopUserManager->flush();
+
+        $this->commandBus->dispatch(new SendAccountRegistrationEmail(
+            $command->email,
+            $command->localeCode,
+            $command->channelCode
+        ), [new DispatchAfterCurrentBusStamp()]);
+
+        if ($channel->isAccountVerificationRequired()) {
+            $token = $this->tokenGenerator->generate();
+            $user->setEmailVerificationToken($token);
+
+            $this->commandBus->dispatch(new SendAccountVerificationEmail(
+                $command->email,
+                $command->localeCode,
+                $command->channelCode
+            ), [new DispatchAfterCurrentBusStamp()]);
+        } else {
             $user->setEnabled(true);
         }
+
```

Comment:
Just a small improvement
```suggestion
        if (!$channel->isAccountVerificationRequired()) {
            $user->setEnabled(true);

            return;
        }
                    
        $token = $this->tokenGenerator->generate();
        $user->setEmailVerificationToken($token);

        $this->commandBus->dispatch(new SendAccountVerificationEmail(
            $command->email,
            $command->localeCode,
            $command->channelCode
        ), [new DispatchAfterCurrentBusStamp()]);
```